### PR TITLE
bug/EUI-6049 fix

### PR DESCRIPTION
**Pull request by Kartheek-S**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-6049

### Change description ###
Hearing response not populate correctly

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/hearings/converters/room-id.answer.converter.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export class RoomIdAnswerConverter implements AnswerConverter {
     return hearingState$.pipe(
       map(state => {
         const hearingResponse = state.hearingRequest.hearingRequestMainModel.hearingResponse;
-        return hearingResponse && hearingResponse.hearingDaySchedule.hearingRoomId;
+        return hearingResponse && hearingResponse.hearingDaySchedule[0].hearingRoomId;
```
Comment (SivaprakasamSivaprakasam) [inline src/hearings/converters/room-id.answer.converter.ts:12]: do we know the hearingdayschedule will be there all the time and if not we may have check hearingDaySchedule.length before pick up the 1st array

File: src/hearings/converters/hearing-response-length.answer.converter.ts [hunk 2]
```diff
@@ -9,10 +9,10 @@ export class HearingResponseLengthAnswerConverter implements AnswerConverter {
   public transformAnswer(hearingState$: Observable<State>): Observable<string> {
     return hearingState$.pipe(
       map(state => {
-        const startTime = moment(state.hearingRequest.hearingRequestMainModel.hearingResponse.hearingDaySchedule.hearingStartDateTime);
-        const endTime = moment(state.hearingRequest.hearingRequestMainModel.hearingResponse.hearingDaySchedule.hearingEndDateTime);
+        const startTime = moment(state.hearingRequest.hearingRequestMainModel.hearingResponse.hearingDaySchedule[0].hearingStartDateTime);
```
Comment (SivaprakasamSivaprakasam) [inline src/hearings/converters/hearing-response-length.answer.converter.ts:12]: do we know the hearingdayschedule will be there all the time and if not we may have check hearingDaySchedule.length before pick up the 1st array

File: src/hearings/resolvers/judicial-user-search-response-resolver.resolve.ts [hunk 3]
```diff
@@ -33,9 +33,11 @@ export class JudicialUserSearchResponseResolver implements Resolve<JudicialUserM
     return this.hearingStore.pipe(select(fromHearingStore.getHearingRequest)).pipe(
       map(hearingRequest => {
         let panelMemberIds: string[] = [];
+        let hearingJudgeIds: string[] = [];
         if (hearingRequest.hearingRequestMainModel && hearingRequest.hearingRequestMainModel.hearingResponse && hearingRequest.hearingRequestMainModel.hearingResponse.hearingDaySchedule) {
-          panelMemberIds = hearingRequest.hearingRequestMainModel.hearingResponse.hearingDaySchedule.panelMemberIds || [];
-          panelMemberIds = [...panelMemberIds, hearingRequest.hearingRequestMainModel.hearingResponse.hearingDaySchedule.hearingJudgeId];
+          panelMemberIds = hearingRequest.hearingRequestMainModel.hearingResponse.hearingDaySchedule.map(member => member.panelMemberId).filter(memberId => memberId) || [];
```
Comment (SivaprakasamSivaprakasam) [inline src/hearings/resolvers/judicial-user-search-response-resolver.resolve.ts:38]: can this panelmember return more than 1 item as all other places we have changed into non-array ?

**Review by SivaprakasamSivaprakasam — COMMENTED** [review]
few comments

File: src/hearings/resolvers/judicial-user-search-response-resolver.resolve.ts (hunk 3 above)
Comment (Kartheek-S) [reply src/hearings/resolvers/judicial-user-search-response-resolver.resolve.ts:38]: hearingDaySchedule is an array. So, getting all paelMenberId and judgeId from hearingDaySchedule[]

File: src/hearings/converters/hearing-response-length.answer.converter.ts (hunk 2 above)
Comment (Kartheek-S) [reply src/hearings/converters/hearing-response-length.answer.converter.ts]: Fixed

File: src/hearings/models/hearingResponse.model.ts [hunk 4]
```diff
@@ -8,6 +8,6 @@ export interface HearingResponseModel {
   laCaseStatus: LaCaseStatus;
   listingStatus: ListingStatus;
   hearingCancellationReason: string;
-  hearingDaySchedule: HearingDayScheduleModel;
+  hearingDaySchedule: HearingDayScheduleModel[];
```
Comment (rayliangatos) [inline src/hearings/models/hearingResponse.model.ts:11]: if we can also amend the /api/hearings/models/hearingResponse.model.ts.model.ts to keep them consistent.

File: src/hearings/models/partyDetails.model.ts [hunk 5]
```diff
@@ -13,4 +13,5 @@ export interface PartyDetailsModel {
   organisationDetails?: OrganisationDetailsModel;
   unavailabilityDOW?: UnavailabilityDOWModel[];
   unavailabilityRanges?: UnavailabilityRangeModel[];
+  hearingSubChannel?: string;
```
Comment (rayliangatos) [inline src/hearings/models/partyDetails.model.ts:16]: same as the api model.

**Review by SivaprakasamSivaprakasam — APPROVED** [review]

**Pull request merged**